### DISCUSSION TITLE
Play back button for audio stimuli

### DISCUSSION THREAD

**Discussion by SearaChen** [jspsych/jsPsych#595] [Q&A] [answered]
How can create a playback button for audio stimuli, there does not seem to be an option in the audio plug-ins ?

Use case: I want the subject to be able to replay the stimuli (an audio) as many times as they want.  

Thank you!

**Comment by jodeleeuw** [comment, marked as answer]
You could set up a looping timeline with the `audio-button-response` plugin so that whenever a repeat button is clicked the trial loops and the audio plays again.

*This code is edited to reflect the bug that @Marita-K-Everhardt caught below!*

```js
var repeat = {
  timeline: [{
    type: 'audio-button-response',
    stimulus: 'sound.mp3',
    choices: ['Play Again', 'Continue'],
  }],
  loop_function: function(data){
    if(data.values[0].button_pressed == 0){
      return true; // loop again!
    } else {
      return false; // continue
    }
  }
}
```

**Reply by EceKayaMPI** [reply]
thanks for the solution but unfortunately it doesn't work :(

**Reply by ghost** [reply]
Try this:

```js
var repeat = {
    timeline: [{
      type: 'audio-button-response',
      stimulus: 'sound.mp3',
      choices: ['Play again','Continue']
    }],
    loop_function: function(data){
      if(data.values()[0].button_pressed == 0){
          return true; // loop again!
      } else {
          return false; // continue
      }
    }
  }
```

**Reply by EceKayaMPI** [reply]
worked straight away! thank you :)

**Reply by jodeleeuw** [reply]
If you are finding this thread later on, note that the solution above will not work for versions 6.3 and later because `button_pressed` was renamed to `response`. Replacing `button_pressed` with `response` should work. See #2440.
